server/src/main/java/org/elasticsearch/plugins/SearchPlugin.java

```diff
@@ -126,11 +127,17 @@
         return emptyList();
     }
     /**
-     * The next {@link Rescorer}s added by this plugin.
+     * The new {@link Rescorer}s added by this plugin.
      */
     default List<RescorerSpec<?>> getRescorers() {
         return emptyList();
     }
+    /**
+     * The new {@link IntervalsSourceProvider}s added by this plugin
+     */
+    default List<IntervalSpec<?>> getIntervalsSourceProviders() {
```

Comment:
Should we provide a scripted intervals source first, I have the feeling that it will cover most of the use cases and that writing a plugin should be for rare cases where the scripted intervals is not enough ?

Reply:
Please keep this.   I will use this right out the gate for functionality I know will never be allowed to be contributed due to the potential performance risk.

Reply:
I'm guessing Matt wants this for wildcard support?  Which wouldn't be covered by a script filter.  I don't think we lose anything by making this pluggable.

Reply:
@romseygeek yes that is my immediate use-case assuming we don't get a wildcard interval into lucene.